From 33cb2616e1024834a7bee40ba3ff404e2b4c1b07 Mon Sep 17 00:00:00 2001
From: SweetCapitan <danil.duhanin@yandex.ru>
Date: Sun, 7 Sep 2025 02:37:12 +0300
Subject: [PATCH 1/3] Add env variables and fix SHORT_BUILD_VERSION for
 Moonraker 0.8

---
 main.py    | 13 ++++++++++---
 printer.py | 10 ++++++++--
 2 files changed, 18 insertions(+), 5 deletions(-)

diff --git a/main.py b/main.py
index d3f4bab..94c017d 100755
--- a/main.py
+++ b/main.py
@@ -4,15 +4,22 @@ import time
 import base64
 from threading import Thread
 from datetime import timedelta
+import io
+import os
 
 from printer import PrinterData
 from lcd import LCD, _printerData
 
+lcd_device = os.getenv('LCD_DEVICE', '/dev/ttyAMA0')
+klippy_sock = os.getenv('KLIPPY_SOCK', '/home/pi/printer_data/comms/klippy.sock')
+moonraker_url = os.getenv('MOONRAKER_URL', '127.0.0.1')
+moonraker_key = os.getenv('MOONRAKER_KEY', 'XXXXXX')
+
 class KlipperLCD ():
     def __init__(self):
-        self.lcd = LCD("/dev/ttyAMA0", callback=self.lcd_callback)
+        self.lcd = LCD(lcd_device, callback=self.lcd_callback)
         self.lcd.start()
-        self.printer = PrinterData('XXXXXX', URL=("127.0.0.1"), klippy_sock='/home/pi/printer_data/comms/klippy.sock', callback=self.printer_callback)
+        self.printer = PrinterData(moonraker_key, URL=moonraker_url, klippy_sock=klippy_sock, callback=self.printer_callback)
         self.running = False
         self.wait_probe = False
         self.thumbnail_inprogress = False
@@ -222,4 +229,4 @@ class KlipperLCD ():
 
 if __name__ == "__main__":
     x = KlipperLCD()
-    x.start()
\ No newline at end of file
+    x.start()
diff --git a/printer.py b/printer.py
index cfbf0bb..8d9bdcf 100644
--- a/printer.py
+++ b/printer.py
@@ -421,7 +421,13 @@ class PrinterData:
 		#alternative approach
 		#full_version = self.getREST('/printer/info')['result']['software_version']
 		#self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
-		self.SHORT_BUILD_VERSION = self.getREST('/machine/update/status?refresh=false')['result']['version_info']['klipper']['version']
+
+        resp = self.getREST('/printer/info')
+        if 'result' in resp and 'software_version' in resp['result']:
+            full_version = resp['result']['software_version']
+            self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
+        else:
+            self.SHORT_BUILD_VERSION = "unknown"
 
 		data = self.getREST('/printer/objects/query?toolhead')['result']['status']
 		#print(json.dumps(data, indent=2))
@@ -668,4 +674,4 @@ class PrinterData:
 		self.setExtTemp(exttemp)
 
 	def setZOffset(self, offset):
-		self.sendGCode('SET_GCODE_OFFSET Z=%s MOVE=1' % offset)
\ No newline at end of file
+		self.sendGCode('SET_GCODE_OFFSET Z=%s MOVE=1' % offset)
-- 
2.47.2


From 580c6e1aacd8d8c5ef00d025cd790cd737da5706 Mon Sep 17 00:00:00 2001
From: SweetCapitan <danil.duhanin@yandex.ru>
Date: Sun, 7 Sep 2025 02:44:07 +0300
Subject: [PATCH 2/3] Add env variables and fix SHORT_BUILD_VERSION for
 Moonraker 0.8

---
 printer.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/printer.py b/printer.py
index 8d9bdcf..ea023a8 100644
--- a/printer.py
+++ b/printer.py
@@ -416,8 +416,7 @@ class PrinterData:
 			print('Web site exists')
 		if self.getREST('/api/printer') is None:
 			return
-		self.update_variable()
-
+	    self.update_variable()
 		#alternative approach
 		#full_version = self.getREST('/printer/info')['result']['software_version']
 		#self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
@@ -429,7 +428,7 @@ class PrinterData:
         else:
             self.SHORT_BUILD_VERSION = "unknown"
 
-		data = self.getREST('/printer/objects/query?toolhead')['result']['status']
+        data = self.getREST('/printer/objects/query?toolhead')['result']['status']
 		#print(json.dumps(data, indent=2))
 		toolhead = data['toolhead']
 		volume = toolhead['axis_maximum'] #[x,y,z,w]
-- 
2.47.2


From 464bc877b6b05b5baef153c5dae4501a1953eade Mon Sep 17 00:00:00 2001
From: SweetCapitan <danil.duhanin@yandex.ru>
Date: Sun, 7 Sep 2025 02:50:23 +0300
Subject: [PATCH 3/3] Add env variables and fix SHORT_BUILD_VERSION for
 Moonraker 0.8

---
 printer.py | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/printer.py b/printer.py
index ea023a8..2277e05 100644
--- a/printer.py
+++ b/printer.py
@@ -416,19 +416,19 @@ class PrinterData:
 			print('Web site exists')
 		if self.getREST('/api/printer') is None:
 			return
-	    self.update_variable()
+		self.update_variable()
 		#alternative approach
 		#full_version = self.getREST('/printer/info')['result']['software_version']
 		#self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
 
-        resp = self.getREST('/printer/info')
-        if 'result' in resp and 'software_version' in resp['result']:
-            full_version = resp['result']['software_version']
-            self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
-        else:
-            self.SHORT_BUILD_VERSION = "unknown"
+		resp = self.getREST('/printer/info')
+		if 'result' in resp and 'software_version' in resp['result']:
+			full_version = resp['result']['software_version']
+			self.SHORT_BUILD_VERSION = '-'.join(full_version.split('-',2)[:2])
+		else:
+			self.SHORT_BUILD_VERSION = "unknown"
 
-        data = self.getREST('/printer/objects/query?toolhead')['result']['status']
+		data = self.getREST('/printer/objects/query?toolhead')['result']['status']
 		#print(json.dumps(data, indent=2))
 		toolhead = data['toolhead']
 		volume = toolhead['axis_maximum'] #[x,y,z,w]
-- 
2.47.2

