diff --git a/main.py b/main.py
index d3f4bab..2323491 100755
--- a/main.py
+++ b/main.py
@@ -4,15 +4,21 @@ import time
 import base64
 from threading import Thread
 from datetime import timedelta
+import io
+import os

 from printer import PrinterData
 from lcd import LCD, _printerData

+lcd_device = os.getenv('LCD_DEVICE', '/dev/ttyAMA0')
+klippy_sock = os.getenv('KLIPPY_SOCK', '/home/pi/printer_data/comms/klippy.sock')
+moonraker_url = os.getenv('MOONRAKER_URL', '127.0.0.1')
+moonraker_key = os.getenv('MOONRAKER_KEY', 'XXXXXX')
+
 class KlipperLCD ():
     def __init__(self):
-        self.lcd = LCD("/dev/ttyAMA0", callback=self.lcd_callback)
+        self.lcd = LCD(lcd_device, callback=self.lcd_callback)
         self.lcd.start()
-        self.printer = PrinterData('XXXXXX', URL=("127.0.0.1"), klippy_sock='/home/pi/printer_data/comms/klippy.sock', callback=self.printer_callback)
+        self.printer = PrinterData(moonraker_key, URL=moonraker_url, klippy_sock=klippy_sock, callback=self.printer_callback)
         self.running = False
         self.wait_probe = False
         self.thumbnail_inprogress = False
@@ -222,4 +228,4 @@ class KlipperLCD ():

 if __name__ == "__main__":
     x = KlipperLCD()
-    x.start()
\ No newline at end of file
+    x.start()
